### PR TITLE
Allow characters in random ticket number.

### PR DESCRIPTION
**Pull request by A-Lawrence**
Allow random numbers to include characters when specified using the @ symbol.

I.e. US-####-@@

c9c1170b63b7fee9f53ad6113cab6dc1

### CODE REVIEW

**Comment by Chefkeks** [comment]
Great! :+1:

**Comment by A-Lawrence** [comment]
@Chefkeks thanks! I figured that I make so many modifications to osTicket that I'd start sharing some of those I feel are more useful!

**Comment by Chefkeks** [comment]
Sharing is great! Hope that @protich and @greezybacon also like it and will implement some of your changes.

File: include/class.sequence.php [hunk 1]
```diff
@@ -104,15 +107,24 @@ function format($format, $number) {
         $number = str_pad($number, $total, $this->padding, STR_PAD_LEFT);
         $output = '';
         $start = $noff = 0;
+        $alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
Comment (greezybacon) [inline include/class.sequence.php]: How should we handle varying alphabets / international characters here?

**Comment by Chefkeks** [comment]
@greezybacon 
I see 3 possibilities here:
- Maybe by adding a new field in the "Manage" popup for the ticket numbering options
- Using the default alphabet from A to Z (maybe not so good for Japanese, Arabic, etc.)
- Leave it as it is right now without characters

**Comment by greezybacon** [comment]
@Chefkeks I too though the sequence manager popup would be a nice place

**Comment by greezybacon** [comment]
I'm tagging this `db-migration` because there is no place in the `sequence` table for an alphabet currently

**Comment by A-Lawrence** [comment]
The database sounds the best place for it.  Our implementation didn't require none ASCII characters, so I never went any further. If I can find the time in the coming weeks, I'll put in some time to add this to the sequence manager too.

**Pull request closed without merging**